src/routes/metadata.ts

```diff
@@ -1,75 +1,112 @@
-const log = require("loglevel");
-const express = require("express");
-const multer = require("multer");
-const { generatePrivate } = require("@toruslabs/eccrypto");
-const { ec: EC } = require("elliptic");
-
-const elliptic = new EC("secp256k1");
+/* eslint-disable security/detect-object-injection */
+import { generatePrivate } from "@toruslabs/eccrypto";
+import { celebrate, Joi, Segments } from "celebrate";
+import { ec as EC } from "elliptic";
+import express, { Request, Response } from "express";
+import log from "loglevel";
+import multer from "multer";
+
+import { getHashAndWriteAsync } from "../database/ipfs";
+import { knexRead, knexWrite } from "../database/knex";
+import redis from "../database/redis";
+import {
+  serializeStreamBody,
+  validateDataTimeStamp,
+  validateGetOrSetNonceSetInput,
+  validateGetOrSetNonceSignature,
+  validateLoopSignature,
+  validateMetadataLoopInput,
+  validateNamespace,
+  validateNamespaceLoop,
+  validateSignature,
+} from "../middleware";
+import { constructKey, getError, MAX_BATCH_SIZE, REDIS_TIMEOUT } from "../utils";
+import { DataInsertType, DBTableName, SetDataInput } from "../utils/interfaces";
 
 const upload = multer({
   limits: { fieldSize: 30 * 1024 * 1024 },
 });
 
-const { getError, constructKey, REDIS_TIMEOUT, MAX_BATCH_SIZE } = require("../utils");
-const {
-  validationMiddleware,
-  validationLoopMiddleware,
-  validateMetadataLoopInput,
-  validateLoopSignature,
-  serializeStreamBody,
-  validateNamespace,
-  validateNamespaceLoop,
-  validateGetOrSetNonceSetInput,
-  validateGetOrSetNonceSignature,
-} = require("../middleware");
-const { knexRead, knexWrite, redisClient: redis, getHashAndWriteAsync } = require("../database");
-const { validateMetadataInput, validateSignature } = require("../middleware");
+const elliptic = new EC("secp256k1");
 
 const router = express.Router();
 
 const NAMESPACES = {
   nonceV2: "noncev2",
   pubNonceV2: "pub_noncev2",
 };
+
 const RESERVED_NAMESPACES = [NAMESPACES.nonceV2, NAMESPACES.pubNonceV2];
 
-router.post("/get", validationMiddleware(["pub_key_X", "pub_key_Y"]), validateNamespace, async (req, res) => {
-  try {
-    const { namespace, pub_key_X: pubKeyX, pub_key_Y: pubKeyY, tableName } = req.body;
-    const key = constructKey(pubKeyX, pubKeyY, namespace);
-    let value;
+const validateSetData = Joi.object({
+  namespace: Joi.string().max(128).required(),
+  pub_key_X: Joi.string().max(66).hex().required(),
+  pub_key_Y: Joi.string().max(66).hex().required(),
+  set_data: Joi.object({
```

Comment:
set_data object should be marked as required.

Reply:
done